src/VSCodeUI.ts

```diff
@@ -53,6 +54,47 @@ export namespace VSCodeUI {
         }
     }
 
+    export function setupEnvironment(terminal: Terminal): void {
+        // do this first so it can be overridden if desired
+        setJavaHomeIfAvailable(terminal);
+
+        type EnvironmentSetting = {
+            environmentVariable: string;
+            value: string;
+        };
+
+        const environmentSettings: EnvironmentSetting[] = workspace.getConfiguration("maven").get("terminal.customEnv");
+        environmentSettings.forEach((s: EnvironmentSetting) => {
+            terminal.sendText(getEnvironmentVariableCommand(s.environmentVariable, s.value), true);
+        });
+    }
+
+    export function setJavaHomeIfAvailable(terminal: Terminal): void {
+        const javaHome: string = workspace.getConfiguration("java").get<string>("home");
```

Comment:
suggest to add comments indicating that the conf `java.home` is from extension `redhat.java`, so as not to confuse future developers.